x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_logic.ts

```diff
@@ -51,6 +58,15 @@ export const FlashMessagesLogic = kea<MakeLogicType<FlashMessagesValues, FlashMe
         clearQueuedMessages: () => [],
       },
     ],
+    toastMessages: [
+      [],
+      {
+        addToastMessage: (toasts, { newToast }) => [...toasts, newToast],
+        dismissToastMessage: (toasts, { removedToast }) =>
+          toasts.filter((toast) => toast.id !== removedToast.id),
```

Comment:
Please note that this is significantly different from `setFlashMessages` in that we are only ever **adding** or **removing** toasts from the array, but never completely overriding/setting the entire array (hence why I avoid the `set` verb in my actions & helpers).

I was generally copying EUI's EuiGlobalToastList example (https://elastic.github.io/eui/#/display/toast - click the "Demo JS" tab) when doing this. I think this approach makes sense because you're supposed to be able to dismiss individual toast messages... But [their guidelines](https://elastic.github.io/eui/#/display/toast/guidelines) do also only say we should only show one toast at time so I'm honestly not 100% sure if I should try to enforce that here 🤷 

Thoughts? Suggestions?

Reply:
With the proposed limit of 5 seconds, there aren't any real cases where we'll have too many stacked toasts, with a few exceptions like the API logs errors. We _could_ try to enforce a max num of toasts at any given time, but I am not sure that adds any value. 

Other than that, I am fine with how you have it here.

I do have a small nit. The Kibana styleguide says to destructure where possible, so I would refactor to this:

```JavaScript
          toasts.filter(({ id }) => id !== removedToast.id),
```